e-ucm/ead
# Exportation to JAR

OK This PR has changed quite a lot. Now it not only addresses saving but the whole exportation process to JAR.

To test this:

1) Before you begin, copy assets/release-sample.json to release.json, if you do not have that file already.
2) Generate engine-with-dependencies.jar file locally using maven: https://github.com/e-ucm/ead/wiki/Common-maven-recipes
3) Make sure theres an engineLibPath in your release.json file pointing to your engine-with-dependencies jar file (e.g.     "engineLibPath": "C:/Users/Javier Torrente/GIT_REPOS/ead/engine/desktop/target/engine-desktop-1.0-SNAPSHOT-jar-with-dependencies.jar")
4) Launch EditorDesktop and open a game.
5) Click File > Export
6) In the dialog, paste the full path of the target destination for your file (e.g. C:\Users\Javier Torrente\Downloads\Export\test.jar)
7) Click OK and wait till the exportation process completes (the UI freezes).

If you double click the exported game jar, it should run!

Note: there are quite a lot of things I've not addressed yet. I'll create separate issues for them:
- Proper exception handling
- Image optimization (use texture atlas)
- Image optimization: remove unused images
- engine-with-dependencies optimization: there are lots of things inside this jar that are not required.
- Give proper feedback to the user on the exportation process
- Avoid the UI freeze while exporting

So please, be good and if you have comments on any of these aspects, just place them into the separate issues : )
